Prox/Prox/Data/PlacesController.swift

```diff
@@ -187,10 +180,20 @@ class PlacesProvider {
         delegate?.placesProvider(self, didUpdatePlaces: displayedPlaces)
     }
 
+    /// Returns a copy of all the places. Callers should not be locked.
+    func getAllPlacesCopy() -> [Place] {
+        return getCopy(ofPlaces: allPlaces)
+    }
+
+    /// Returns a copy of the displayed places. Callers should not be locked.
     func getDisplayedPlacesCopy() -> [Place] {
+        return getCopy(ofPlaces: displayedPlaces)
+    }
+
+    private func getCopy(ofPlaces places: [Place]) -> [Place] {
         var placesCopy: [Place] = []
         placesLock.withReadLock {
-            placesCopy = Array(self.displayedPlaces)
+            placesCopy = Array(places)
```

Comment:
Thought: this means that since `allPlaces`/`displayedPlaces` are protected fields and Swift doesn’t have a concept of atomic assignment, every time we call a method with places as an argument, it has to be locked.

Otherwise, in theory, we could be in the middle of copying (for a fn call) when someone else modifies the Array.

(talking on Slack to figure out if this is actually an issue)